static/src/javascripts/projects/common/modules/onward/history.js

```diff
@@ -0,0 +1,477 @@
+};
+
+const mostViewedSeries = (): string => {
+    const summary = seriesSummary();
+    let series = '';
+
+    Object.keys(summary).forEach(key => {
+        const views = summary[key];
+
+        if (!series || views > summary[series]) {
+            series = key;
+        }
+    });
+
+    return series;
+};
+
+const deleteFromSummary = (tag: string): void => {
+    const summary = getSummary();
+
+    delete summary.tags[tag];
+    saveSummary(summary);
+};
+
+const isRevisit = (pageId: string): boolean => {
+    const visited = getHistory().find(page => page[0] === pageId);
+
+    return !!(visited && visited[1] > 1);
+};
+
+const pruneSummary = (summary: Object, mockToday?: ?number) => {
+    const newToday = mockToday || today;
+    const updateBy = newToday - summary.periodEnd;
+
+    if (updateBy !== 0) {
+        summary.periodEnd = newToday;
+
+        Object.keys(summary.tags).forEach(tid => {
+            const record = summary.tags[tid];
+
+            const result = buckets.map(bucket => {
+                if (record[bucket.indexInRecord]) {
+                    const visits = record[bucket.indexInRecord]
+                        .map(day => {
+                            const newAge = day[0] + updateBy;
+                            return newAge < summaryPeriodDays && newAge >= 0
+                                ? [newAge, day[1]]
+                                : false;
+                        })
+                        .filter(Boolean);
+
+                    return visits.length > 1 ||
+                        (visits.length === 1 &&
+                            visits[0][0] < forgetUniquesAfter)
+                        ? visits
+                        : [];
+                }
+
+                return [];
+            });
+
+            if (result.some(r => r.length)) {
+                summary.tags[tid] = [record[0]].concat(result);
+            } else {
+                delete summary.tags[tid];
+            }
+        });
+    }
+
+    return summary;
+};
+
+const tally = (
+    visits: Array<Array<number>>,
+    weight: number = 1,
+    minimum: number = 1
+): number => {
+    let totalVisits = 0;
+
+    const result = visits.reduce((t, day) => {
+        const dayOffset = day[0];
+        const dayVisits = day[1];
+
+        totalVisits += dayVisits;
+        return t + weight * (9 + dayVisits) * (summaryPeriodDays - dayOffset);
+    }, 0);
+
+    return totalVisits < minimum ? 0 : result;
+};
+
+const getPopular = (opts: ?Object): Array<Array<string>> => {
+    const tags = getSummary().tags;
+    let tids = Object.keys(tags);
+
+    const op = Object.assign(
+        {},
+        {
+            number: 100,
+            weights: {},
+            thresholds: {},
+        },
+        opts
+    );
+
+    tids = op.whitelist
+        ? tids.filter(tid => op.whitelist.indexOf(tid) > -1)
+        : tids;
+    tids = op.blacklist
+        ? tids.filter(tid => op.blacklist.indexOf(tid) === -1)
+        : tids;
```

Comment:
Sure, it's a little easier to read than the conditional ternary operator, and `includes` is preferable to `indexOf`!